### PR TITLE
Dev - now with CI/CD pipeline!

### PR DESCRIPTION
**Pull request by sofiiagran**
## Description

In this PR there is a new workflow!

## Changes

The workflow for dev remains the same.

However, now there is a specific workflow that only runs on PR to staging, called build.yml
* In this workflow there is a CI pipeline that makes sure the code is tested and can be built.

There is also a specific workflow that only runs on PR to main, called deploy.yml
* In this workflow there is a CI/CD pipeline that deploy the app to the play store automatically.

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

Proof that the CI/CD works:
Skærmbillede 2023-11-30 kl  18 00 00
Skærmbillede 2023-11-30 kl  17 59 45

## Notes for Reviewers

One cannot see the actual CI pipeline in this PR, it will not be called before one creates a merge to stage.
The same goes for the CI/CD pipeline for main, it will be visible when we create PR to main. 
However, everything has been tested as one can see in the screen shots. 

The app can now be deployed automatically.
There are some limitations:
* The version in the app.json has to be changed manually before the PR
* Expo/EAS does not support automatic release notes, therefore, one has to log in to the play store console and update the release note manually after deploying. A release note is not required, however, it is best practice to add one.

### CODE REVIEW

**Comment by Hifzza** [comment]
Great job on the pull request! The screenshots serve as compelling evidence of the functionality, reassuring us that the system works seamlessly, even if the CI pipeline isn't immediately visible in the PR. 👍

**Review by amaliedilling — APPROVED** [review]

**Review by MortenMunk — APPROVED** [review]
I am a simple man..
I see an image of a nice pipeline, i approve 🤷‍♂️

File: eml/app.json [hunk 1]
```diff
@@ -15,7 +14,7 @@
     ],
     "name": "Educado",
     "slug": "educado-mobile",
-    "version": "1.0.0",
+    "version": "2.0.1",
```
[the remark this answers is not in the document]
Comment (sofiiagran) [reply eml/app.json:17]: well it is a new version tho, in comparison to the other one

**Review by TheMOHD3886 — APPROVED** [review]
looks aight 👍

**Pull request merged**